jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/table/DataTableAssert.java

```diff
@@ -27,154 +27,179 @@
+            if (!condition.execute(allRows.get(i - 1), allRows.get(i)))
+                jdiAssert("Table not sorted at " + i + " row", Matchers.is(""), messages);
+        jdiAssert("Table is sorted", Matchers.is("Table is sorted"), messages);
         return this;
     }
-
+    
     public Compare exact(int count) {
         return new Compare(count, this, EXACT);
     }
+    
     public Compare atLeast(int count) {
         return new Compare(count, this, ATLEAST);
     }
+    
     public Compare no() {
         return exact(0);
     }
+    
     public Compare all() {
         return new Compare(this);
     }
+    
     public Compare onlyOne() {
         return exact(1);
     }
-
+    
     public enum CompareType {
         EXACT("exactly"), ATLEAST("at least"), ALL("all");
         public String text;
-        CompareType(String text) { this.text = text; }
+        
+        CompareType(String text) {
+            this.text = text;
+        }
     }
-
+    
     public class Compare implements JAssert {
-
+        
         public int count;
         public String name;
         public String printText;
         DataTableAssert<L, D> dtAssert;
         CompareType compareType;
-        public JDIBase base() { return DataTableAssert.this.base(); }
-
+        
+        public JDIBase base() {
+            return DataTableAssert.this.base();
+        }
+        
         private Compare(DataTableAssert<L, D> dtAssert) {
             this.dtAssert = dtAssert;
             this.compareType = ALL;
             this.printText = "all rows";
             this.name = dtAssert.name;
         }
+        
         private Compare(int count, DataTableAssert<L, D> dtAssert, CompareType compareType) {
             this.count = count;
             this.dtAssert = dtAssert;
             this.compareType = compareType;
             this.printText = format("has %s %s rows that", compareType.text, count);
             this.name = dtAssert.name;
         }
-
+        
         /**
          * Check that the table has rows that meet expected condition
+         *
          * @param condition to compare
          * @return DataTableAssert
          */
         @JDIAction("Assert that '{name}' {printText} meet expected condition")
-        public DataTableAssert<L, D> rows(JFunc1<D, Boolean> condition) {
+        public DataTableAssert<L, D> rows(JFunc1<D, Boolean> condition, String ... messages) {
             switch (compareType) {
                 case EXACT:
-                    jdiAssert(table().dataRows(condition), hasSize(count));
+                    jdiAssert(table().dataRows(condition), hasSize(count), messages);
                     break;
                 case ALL:
                     List<D> rows = table().allData();
-                    jdiAssert(LinqUtils.all(rows, condition), Matchers.is(true));
+                    jdiAssert(LinqUtils.all(rows, condition), Matchers.is(true), messages);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5133577)